experimental/NeuralNets/Lib.idr

```diff
@@ -0,0 +1,151 @@
+module Lib
+import Prelude.Doubles as db
+
+export
+stopwords : List String
+stopwords = ["the", "a", "and", "is", "be", "will", "is"] 
+
+export
+nums : String
+nums = "0123456789!@#$%^&*()_+=-:;"
+
+export
+removeStopWords : String -> List String
+removeStopWords line = do
+              let tmp = words line
+              foldr (\x, acc => if elem x stopwords then acc else x::acc ) [] tmp
```

Comment:
@Eman22S, I'm not sure the usage of `do` and `let` is generally appropriate for code without side effect like this one.  Using `where` or something instead might be more appropriate/idiomatic (@sroberti might have thoughts on this).

Also, you can simplify that line by using `filter` or list comprehension instead of `foldr`.  If you use `filter` you can also probably avoid using lambda by combining `flip` and partial evaluation.

Reply:
Yes. `Do` can be replaced by `let ..in`. The initial implementation was using `filter` . I used `foldr` here and other places since `foldr` can minimize the cost of using `filter` and `reverse` later on. since inserting elements using `:` reverses the order of initial order.  `filter ` and `reverse` are both `O(n)` .  Also by partial evaluation , do you mean something like curried functions? I tried using them without much lack. Clearly it's different from how haskell uses curried functions.(Idris requires type declaration where as haskell do not) @ sroberti please share your thoughts if you have any ideas.

Reply:
I don't understand, are you saying that `filter` would be slower than your use of `foldr`?

Looking at its implementation I don't believe it would (but I didn't try)
https://github.com/idris-lang/Idris2/blob/299a31de5bd64a584a2f7d9090eb2f5680efaeac/libs/base/Data/List.idr#L107-L112

Reply:
What I had in mind was
```idris
removeStopWords line = filter ((flip elem) stopwords) (words line)
```
Wouldn't that be semantically equivalent and as fast?

Reply:
Sorry for not responding sooner ..holy month has been getting me busy lately anyways. I think I am going with the mine because yours return simply the stop words which is not what we want. removestopwords should return the text weeding out stop words for example. The prince is the Future king will be [prince, future, king]. Yours return [the, is] which we don't want. I have removed the do clause however as it is indeed unnecessary

Reply:
Oh right, it would be something like
```idris
removeStopWords line = filter (not . ((flip elem) stopwords)) (words line)
```
then.

Reply:
For sake of completion here's the version using `filter` and `lambda`
```idris
removeStopWords line = filter (\x => not (elem x stopwords)) (words line)
```
as well as the version using list comprehension
```
removeStopWords line = [x | x <- (words line), not (elem x stopwords))]
```